### PR TITLE
Offline Mode: Previews

### PR DESCRIPTION
**Pull request by kean**
Update "Preview" button to use the new services.

### Previews

**Test 1.1**

- Open an existing post (publish|publishPrivate|scheduled)
- Make a change 
- Tap "Preview"
- **Verify** that the preview was generated
- Open the same post on the web
- **Verify** that the "View autosave" panel is shown 

> Note: not ideal, but it matches the existing behavior in production

**Test 1.2**

- Open an existing draft post
- Make a change
- Tap "Preview"
- **Verify** that the preview was generated
- **Verify** that the post was synced

**Test 1.3**

- Create a new draft post
- Tap "Preview"
- **Verify** that the preview was generated
- **Verify** that the post was synced

### Misc

**Test 2.1**

**Precondition**: app is offline.

- Open an existing draft post with content A
- Change the content to B
- Save draft
- Change the content back to A
- Save draft
- **Verify** that the spinner was dismissed; not network requests made

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/PostEditor+MoreOptions.swift [hunk 1]
```diff
@@ -31,6 +31,42 @@ extension PostEditor {
...
+            completion(nil, nil)
+            return
+        }
+
+        Task { @MainActor in
+            let coordinator = PostCoordinator.shared
+            do {
+                if post.isStatus(in: [.draft, .pending]) {
+                    SVProgressHUD.setDefaultMaskType(.clear)
+                    SVProgressHUD.show(withStatus: Strings.savingDraft)
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Post/PostEditor+MoreOptions.swift:48]: I slightly changed the copy to communicate to the user that we're saving the changes.

Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Post/PostEditor+MoreOptions.swift:48]: Nice touch - makes it clear

File: WordPress/WordPressTest/PostCoordinatorSyncTests.swift [hunk 2]
```diff
@@ -457,6 +457,83 @@ class PostCoordinatorSyncTests: CoreDataTestCase {
         XCTAssertEqual(post.content, "content-c")
         XCTAssertEqual(post.revision, revision3)
     }
+
+    /// Scenario: the app needs to upload the most recent revision to the server
+    /// to generate a preview.
+    func testSaveDraftPostChangesImmediatelly() async throws {
```
Comment (momo-ozawa) [inline WordPress/WordPressTest/PostCoordinatorSyncTests.swift]:
```suggestion
    func testSaveDraftPostChangesImmediately() async throws {
```

File: WordPress/WordPressTest/PostCoordinatorSyncTests.swift [hunk 3]
```diff
@@ -457,6 +457,83 @@ class PostCoordinatorSyncTests: CoreDataTestCase {
...
+        try await coordinator._save(post)
+
+        // THEN all changes were synced
+        XCTAssertEqual(post.content, "content-c")
+        XCTAssertNil(post.revision)
+    }
+
+    /// Scenario: the app needs to upload the most recent revision to the server
+    /// to generate a preview.
+    func testSaveDraftPostChangesImmediatellyFailure() async throws {
```
Comment (momo-ozawa) [inline WordPress/WordPressTest/PostCoordinatorSyncTests.swift]:
```suggestion
    func testSaveDraftPostChangesImmediatelyFailure() async throws {
```

**Review by momo-ozawa — APPROVED** [review]
LGTM

**Pull request merged**